python/chronos/src/bigdl/chronos/forecaster/autoformer_forecaster.py

```diff
@@ -0,0 +1,230 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import torch
+from bigdl.chronos.forecaster.base_forecaster import BasePytorchForecaster
+from bigdl.chronos.model.autoformer import model_creator
+from torch.utils.data import TensorDataset, DataLoader
+from bigdl.chronos.model.autoformer.Autoformer import AutoFormer
+import torch.nn as nn
+import numpy as np
+import os
+
+
+class AutoformerForecaster(BasePytorchForecaster):
+    def __init__(self,
+                 seq_len,
+                 label_len,
+                 pred_len,
+                 enc_in,
+                 freq,
+                 dec_in,
+                 c_out,
+                 output_attention=False,
+                 moving_avg=25,
+                 d_model=512,
+                 embed='timeF',
+                 dropout=0.05,
+                 factor=3,
+                 n_head=8,
+                 d_ff=2048,
+                 activation='gelu',
+                 e_layer=2,
+                 d_layers=1):
+        # config setting
+        self.config = {
+            "seq_len": seq_len,
+            "label_len": label_len,
+            "pred_len": pred_len,
+            "output_attention": output_attention,
+            "moving_avg": moving_avg,
+            "enc_in": enc_in,
+            "d_model": d_model,
+            "embed": embed,
+            "freq": freq,
+            "dropout": dropout,
+            "dec_in": dec_in,
+            "factor": factor,
+            "n_head": n_head,
+            "d_ff": d_ff,
+            "activation": activation,
+            "e_layer": e_layer,
+            "c_out": c_out,
+            "d_layers": d_layers
+        }
+
+        self.distributed = False
+        self.seed = None
+        self.checkpoint_callback = False
+        current_num_threads = torch.get_num_threads()
+        self.num_processes = max(1, current_num_threads//8)
+        self.use_ipex = False
+        self.onnx_available = True
+        self.quantize_available = True
+        self.use_amp = False
+
+        self.model_creator = model_creator
+        self.internal = model_creator(self.config)
+
```

Comment:
it there any possibility we move these changes to base_forecaster, which means making base forecaster support a more generalized input?